### PR TITLE
Updated rst files to refer to 16.0 release

### PR DESCRIPTION
**Pull request by gcomoretto**
Please have a look to these (trivial) changes.. I hope I have no more 15.0 laying around...

Before merging this branch, we should merge John's branch:
https://github.com/lsst/pipelines_lsst_io/tree/u/swinbank/16.0-tickets

### CODE REVIEW

File: install/demo.rst [hunk 1]
```diff
@@ -28,8 +28,8 @@ Then download the demo's data (if you aren't running the current stable release,
 
 .. code-block:: bash
 
-   curl -L https://github.com/lsst/lsst_dm_stack_demo/archive/15.0.tar.gz | tar xvzf -
-   cd lsst_dm_stack_demo-15.0
+   curl -L https://github.com/lsst/lsst_dm_stack_demo/archive/15.6.tar.gz | tar xvzf -
```
Comment (jdswinbank) [inline install/demo.rst]: `15.6`? Should be `16.0`?

Comment (gcomoretto) [reply install/demo.rst]: Yes, I will correct it.

**Comment by jonathansick** [comment]
@gcomoretto can you merge #91 and then rebase this branch on `16.0`. Then I'll take a look.

**Comment by jonathansick** [comment]
What did we want to do about?

> It is based on 2018 weekly 10.

https://pipelines.lsst.io/v/DM-14807/releases/v16_0.html

I think this should be weekly 22?

So I think either say

```rst
It is based on the ``w_2018_22`` weekly release.
```

or delete the sentence altogether.

**Comment by jonathansick** [comment]
https://pipelines.lsst.io/v/DM-14807/releases/notes.html#summer-2018-release-v16-0

> This is the first major release that do not supports Python 2.

Should be:

> This is the first major release that does not support Python 2.

**Comment by jonathansick** [comment]
I've got the Python 2.7 removal in #94.

**Comment by gcomoretto** [comment]
I have implemented the changes.

**Pull request merged**